apps/beeswax/src/beeswax/server/dbms_tests.py

```diff
@@ -149,6 +149,28 @@ def test_get_hive_ha_discovery_all_server_down(self):
                 assert_equal(e.message, 'There is no running Hive server available')
 
 
+  def test_get_hs2_discovery(self):
+
+    with patch('beeswax.conf.HIVE_DISCOVERY_HS2.get') as HIVE_DISCOVERY_HS2:
+      with patch('beeswax.conf.HIVE_DISCOVERY_HIVESERVER2_ZNODE.get') as HIVE_DISCOVERY_HIVESERVER2_ZNODE:
+        with patch('beeswax.server.dbms.KazooClient') as KazooClient:
+          HIVE_DISCOVERY_HS2.return_value = True
+          HIVE_DISCOVERY_HIVESERVER2_ZNODE.return_value = True
+          KazooClient.return_value = Mock(
+            exists=Mock(return_value=True), # Bug "TypeError: expected string or buffer" if False, to add a new test case and fix
+            get_children=Mock(return_value=['serverUri=hive-llap-1.gethue.com:10000;serverUri=hive-llap-2.gethue.com:10000'])
```

Comment:
Just for the record (not 100% sure), this might be shorter this way:

KazooClient.get_children.return_value = ['serverUri=hive-llap-1.gethue.com:10000;serverUri=hive-llap-2.gethue.com:10000']
etc